### PR TITLE
Expose context within constructors and operations

### PR DESCRIPTION
**Pull request by skygering**
I have exposed the context argument in the type constructors and within the geos operations. This addresses part of issue #91. All current tests pass so there is no regression. I also added one test per type constructor. If we think we want/need tests, let me know.

### CODE REVIEW

File: src/geos_operations.jl
Comment (yeesian) [inline src/geos_operations.jl:132]: Consider matching the style of the function signatures in the rest of this file. I.e. for single line declarations, it'll be 
```julia
topologyPreserveSimplify(obj::Geometry, tol::Real, context::GEOSContext = _context) =
```
without any linebreaks.

(If you do this, there are other occurrences in this file.)

**Review by yeesian — APPROVED** [review]
Thank you, as always! Just a minor nit that I'll wait for your response (or other reviewers; if any) before merging.

File: src/geos_operations.jl
Comment (skygering) [reply src/geos_operations.jl]: Good point! I was struggling because I had line wrap on because I didn't want the lines to be too long but I don't think it is worth the lack of readability. I tried to standardize the style a bit more.

Comment (yeesian) [reply src/geos_operations.jl]: Thank you!

Maybe we can set this package up with https://github.com/domluna/JuliaFormatter.jl (like in [ArchGDAL.jl](https://github.com/yeesian/ArchGDAL.jl/blob/master/.github/workflows/format_pr.yml)); then we won't have to nitpick on style guides in pull requests and still get e.g. [`margin = 80`](https://github.com/yeesian/ArchGDAL.jl/blob/66066333e7a6c06f3a1086ec1de0b7d4ac3ca79c/.JuliaFormatter.toml#L6) done for us automatically.

**Pull request merged**
